cms/src/components/languages/CreateOrEditLanguageModal.vue

```diff
@@ -0,0 +1,147 @@
+<script setup lang="ts">
+import { ref, computed, watch } from "vue";
+import { db, DocType, type LanguageDto } from "luminary-shared";
+import LInput from "@/components/forms/LInput.vue";
+import LButton from "@/components/button/LButton.vue";
+import GroupSelector from "../groups/GroupSelector.vue";
+import * as _ from "lodash";
+
+// Props for visibility and language to edit
+type Props = {
+    isVisible: boolean;
+    language?: LanguageDto;
+};
+const props = defineProps<Props>();
+
+// Track the previous state for dirty checking
+const previousLanguage = ref<LanguageDto | null>(null);
+
+// Emit events to close the modal and trigger creation or update
+const emit = defineEmits(["close", "created", "updated"]);
+
+// Check if we are in edit mode (if a language is passed)
+const isEditMode = computed(() => !!props.language);
+
+// New language or edited language object
+const newLanguage = ref<LanguageDto>({
+    _id: db.uuid(), // Generate new ID for create mode
+    name: "",
+    languageCode: "",
+    memberOf: ["group-languages"],
```

Comment:
We should not hardcode a default group as this group might not exist in a production system.